cgp/local_search/random_search.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+from typing import Callable
+
+
+class RandomLocalSearch:
+    def __init__(
+        self,
+        objective: Callable,
+        seed: int,
+        n_steps: int = 10,  # todo: should this be dependent on the number of parameters?
+        permissible_values: np.ndarray = np.logspace(start=0, stop=7, num=8, base=2, dtype=int),
+    ) -> None:
+        self.objective = objective
+        self.seed = seed
+        self.n_steps = n_steps
+        self.permissible_values = permissible_values
+
+    def __call__(self, ind) -> None:
+        rng = np.random.RandomState(self.seed)
```

Comment:
this can be moved into the `if len(` block, so that it's not unnecessarily executed for individuals without parameters